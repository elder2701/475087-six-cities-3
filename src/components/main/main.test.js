import React from "react";
import renderer from "react-test-renderer";
import Main from "./main.jsx";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import createMapBlock from "../map/create-map-block.js";

const places = [
  `Beautiful & luxurious apartment at great location`,
  `Canal View Prinsengracht`,
  `Wood and stone place`,
  `Nice, cozy, warm big bed apartment`
];
const city = ``;
const mockStore = configureStore([]);

const offers = [
  {
    id: 2,
    figure: `img/room.jpg`,
    mark: ``,
    price: 8220,
    priceText: `night`,
    rating: 2,
    name: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    coordinates: [53.369553943508, 5.85309666406198]
  },
  {
    id: 10,
    figure: `img/room.jpg`,
    mark: ``,
    price: 101320,
    priceText: `night`,
    rating: 5,
    name: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    coordinates: [53.369553943508, 5.85309666406198]
  },
  {
    id: 3,
    figure: `img/room.jpg`,
    mark: `Premium`,
    price: 83210,
    priceText: `night`,
    rating: 1,
    name: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    coordinates: [53.369553943508, 5.85309666406198]
  },
  {
    id: 4,
    figure: ``,
    mark: ``,
    price: 8210,
    priceText: `night`,
    rating: 4,
    name: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    coordinates: [53.369553943508, 5.85309666406198]
  }
];

it(`<Main /> sould be render`, () => {
  const store = mockStore({});
  createMapBlock();
  const tree = renderer
    .create(
        <Provider store={store}>
          <Main city={city} places={places} optionSorting="" cityOffers={offers} />
        </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
