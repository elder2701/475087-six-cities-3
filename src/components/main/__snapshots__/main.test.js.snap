// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> sould be render 1`] = `
<main
  className="page__main page__main--index"
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <div
    className="tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
            onClick={[Function]}
          >
            <span>
              Amsterdam
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>
  <div
    className="cities"
  >
    <div
      className="cities__places-container container "
    >
      <section
        className="cities__places places"
      >
        <h2
          className="visually-hidden"
        >
          Places
        </h2>
        <b
          className="places__found"
        >
          2
           places to stay in 
          Amsterdam
        </b>
        <form
          action="#"
          className="places__sorting"
          method="get"
        >
          <span
            className="places__sorting-caption"
          >
            Sort by
          </span>
          <span
            className="places__sorting-type"
            onClick={[Function]}
            tabIndex="0"
          >
            
            <svg
              className="places__sorting-arrow"
              height="4"
              width="7"
            >
              <use
                xlinkHref="#icon-arrow-select"
              />
            </svg>
          </span>
          <ul
            className="places__options places__options--custom"
          >
            <li
              className="places__option"
              onClick={[Function]}
              tabIndex="0"
            >
              Popular
            </li>
            <li
              className="places__option"
              onClick={[Function]}
              tabIndex="0"
            >
              Price: low to high
            </li>
            <li
              className="places__option"
              onClick={[Function]}
              tabIndex="0"
            >
              Price: high to low
            </li>
            <li
              className="places__option"
              onClick={[Function]}
              tabIndex="0"
            >
              Top rated first
            </li>
          </ul>
        </form>
        <div
          className="cities__places-list tabs__content places__list"
        >
          <article
            className="cities__place-card place-card"
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              className="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src=""
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    467
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    In bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "20%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
                onClick={[Function]}
              >
                <a
                  href="#"
                >
                  Amazing and Extremely Central Flat
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                apartment
              </p>
            </div>
          </article>
          <article
            className="cities__place-card place-card"
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <div
              className="cities__image-wrapper place-card__image-wrapper"
            >
              <a
                href="#"
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src=""
                  width="260"
                />
              </a>
            </div>
            <div
              className="place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    133
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button"
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    In bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "60%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
                onClick={[Function]}
              >
                <a
                  href="#"
                >
                  Waterfront with extraordinary view
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                room
              </p>
            </div>
          </article>
        </div>
      </section>
      <div
        className="cities__right-section"
      >
        <section
          className="cities__map"
          id="map"
        />
      </div>
    </div>
  </div>
</main>
`;
